### PR TITLE
Add Illuminate\Database\Query\Expression to stubs

### LINKED ISSUE #784: Parameter $column of method orWhere() errors when Illuminate\Database\Query\Expression is given
- Larastan Version: 0.7.0
- `--level` used: 5
- PHPStan Version: 0.12.75

### Description

When on level 5, the `orWhere()` method on `Illuminate\Database\Eloquent\Builder<Illuminate\Database\Eloquent\Model>::orWhere()` fails when a `Illuminate\Database\Query\Expression` is given, even though it's in the DocBlock of aforementioned method.

```php
/**
 * Add an "or where" clause to the query.
 *
 * @param  \Closure|array|string|\Illuminate\Database\Query\Expression  $column
 * @param  mixed  $operator
 * @param  mixed  $value
 * @return $this
 */
public function orWhere($column, $operator = null, $value = null)
```

### Laravel code where the issue was found

```php
$subQuery->orWhere(
    DB::raw(sprintf('lower(%s)', $column)),
    'like',
    '%' . strtolower($searchTerm) . '%'
);
```
### Full error:

```json
{
    "description": "Parameter #1 $column of method Illuminate\\Database\\Eloquent\\Builder<Illuminate\\Database\\Eloquent\\Model>::orWhere() expects array<int|string, mixed>|Closure|string, Illuminate\\Database\\Query\\Expression given.",
    "fingerprint": "3d188e1afc222590ee964c4330945aaa601e2139aaf0454b43c2e3ad64662aef",
    "location": {
        "path": "app/JsonApi/AbstractAdapter.php",
        "lines": {
            "begin": 46
        }
    }
}
```

Note: it says it expects an array that looks like `int|string` or `mixed`, that's another error I think?

### PR DESCRIPTION
**Pull request by ttomdewit**
- [x] Added or updated tests
- [ ] Documented user facing changes
- [x] Updated CHANGELOG.md

Resolves #784

**Changes**

Adds the `Illuminate\Database\Query\Expression` as DocBlock parameter to the `orWhere()` stub (more stubs will follow once this is in an acceptable and tested state).

**Breaking changes**

I don't believe anything breaks with these changes?

### CODE REVIEW

**Comment by szepeviktor** [comment]
1. Could you add this to all methods @canvural mentioned?
2. Your git client is set up with a non-GitHub email address.
3.

**Comment by ttomdewit** [comment]
> 1. Could you add this to all methods @canvural mentioned? https://github.com/laravel/framework/pull/34828/files
> 2. Your git client is set up with a non-GitHub email address. beach_umbrella

Yes, I will definitely add the missing stubs once this `orWhere()` method is in an acceptable state (so it's tested and you're willing to merge it). At that point I know and understand where to find and update (or add missing) tests. At this point I haven't the faintest idea how to test my change(s).

**Comment by szepeviktor** [comment]
Maybe in model-property-builder.php?

https://github.com/nunomaduro/larastan/blob/073778dc8f22de6be70ea66a2ce984c524c9f108/tests/Rules/Data/model-property-builder.php#L25-L29

**Comment by ttomdewit** [comment]
> Maybe in model-property-builder.php?
> 
> https://github.com/nunomaduro/larastan/blob/073778dc8f22de6be70ea66a2ce984c524c9f108/tests/Rules/Data/model-property-builder.php#L25-L29

Yeah that's where I looked as well but can't get it to pass when I add a Query Expression method.

**Comment by ttomdewit** [comment]
> Maybe in model-property-builder.php?
> 
> https://github.com/nunomaduro/larastan/blob/073778dc8f22de6be70ea66a2ce984c524c9f108/tests/Rules/Data/model-property-builder.php#L25-L29

I'm currently looking at `\Tests\Rules\ModelPropertyRuleTest::testModelPropertyRuleOnBuilder` and how I can update it to accept a Query Expression, but no luck so far. If I succeed I've edited all files that originally tested against `orWhere()`.

**Comment by ttomdewit** [comment]
While adding the missing DocBlock information I've found a missing method: https://github.com/laravel/framework/blob/ba801e339af7e31bf1bb06414636601d61473ab9/src/Illuminate/Database/Eloquent/Builder.php#L1274

**Comment by ttomdewit** [comment]
I'm going to leave it here and I'll get back to it at a later time.

File: tests/Features/ReturnTypes/BuilderExtension.php
Comment (canvural) [inline tests/Features/ReturnTypes/BuilderExtension.php:170]: I think the test in the `tests/Features/Methods/ModelExtension.php` file is enough. We can remove this.

**Review by canvural — CHANGES_REQUESTED** [review]
Hi,

All looks good! When you add tests for all the methods, we can merge :+1:

File: tests/Features/ReturnTypes/BuilderExtension.php
Comment (ttomdewit) [reply tests/Features/ReturnTypes/BuilderExtension.php:170]: I'll make sure to update tomorrow or during my weekend!

**Comment by ttomdewit** [comment]
> Hi,
> 
> 
> 
> All looks good! When you add tests for all the methods, we can merge :+1: 

Cheers! I'll make sure to add them tomorrow or during my weekend.

**Comment by ttomdewit** [comment]
@szepeviktor @canvural May I request a new review? I think I've added all missing tests.

**Comment by ttomdewit** [comment]
> 1. Could you add this to all methods @canvural mentioned? https://github.com/laravel/framework/pull/34828/files
> 2. Your git client is set up with a non-GitHub email address. beach_umbrella

For the second issue I've added my work email account that was setup in `git config user.email` and attached it to my personal GitHub account. I didn't know that was an option. All commits should now be attached to my credentials.

**Review by szepeviktor — APPROVED** [review]
I blindly approve anything that passes CI :)

**Comment by ttomdewit** [comment]
@canvural Hi! Don't want to be rude or impatient but are we waiting for your review before merging? I appreciate you're busy so please take your time, I don't want to sound or act entitled.

**Comment by szepeviktor** [comment]
@canvural Could you click a button here?

**Comment by canvural** [comment]
Sorry for the delay. It looks good :+1: Thank you!

**Comment by ttomdewit** [comment]
Thanks!

**Pull request merged**
